db/db_impl/db_impl_open.cc

```diff
@@ -935,6 +936,22 @@ Status DBImpl::RecoverLogFiles(const std::vector<uint64_t>& wal_numbers,
         }
         if (stop_replay_for_corruption) {
           logFileDropped();
+          // Skip over any subsequent empty wal files and update corrupted wal
+          // number. This avoids false corruption error due to some column
+          // families pointing to empty wal file next to the corrupted wal
+          // file.
+          ++ii;
```

Comment:
Sorry it isn't obvious to me why we increment `ii` here. I thought we executed this logic only for WALs after the corrupted WAL, so we are already past it. Also it seems to execute for all non-empty WALs following the first corrupted WAL, in which case could this scenario increase `corrupted_wal_number` too far?

- WAL 1 is corrupt
- WAL 2 is empty
- WAL 3 is empty
- WAL 4 is non-empty
- WAL 5 is empty

I am not sure what prevents `corrupted_wal_number` from reaching 5, but feel it should be 3.